### PR TITLE
Issue #13345: Enable examples tests for CommentIndentationCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for CommentIndentationCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **CommentIndentationCheck** in the project. The implementation focuses on ensuring that CommentIndentation checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Review by romani — CHANGES_REQUESTED** [review]
Please move `To configure the Check:` to the top of examples.

Please add one more example with different `tokens`. To show how behavior is different if it is customized.

Two top examples should be same by structure, different violations. All other examples ok to be different for now.
Put in example1 single line comments, to show that changes in tokens can allow single line comments or multiline to be skipped from validation.

As second example will be with custom config, but all other examples are for default.
We will need to print default config one more time in page and provide wording that all below examples for default config.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example4.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  public void foo1() {
+    // comment
+    // block
+    // it is OK (we cannot clearly detect user intention of explanation target)
+  }
+
+  public void foo2() {
+ // comment
+ // block
+ // violation below, 'Comment has incorrect indentation level 1, expected is 2'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example4.java:21]: Please put violation comments a bit far from target comments to not confuse readers what is code validating and what is a hint as result bof validation

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+package com.puppycrawl.tools.checkstyle.checks.indentation.commentsindentation;
+
+// xdoc section -- start
+public class Example3 {
+  /////////////////////////////// it is OK
+
+  public void foo1() {
+    int a = 0;
+  }
+  // violation below, 'Comment has incorrect indentation level 4, expected is 2'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java:17]: Please move it above `foo2()` and make it `violation 2 lines above `

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example1.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// xdoc section -- start
+public class Example1 {
+  void testMethod() {
+    /*
+     * it is Ok
+     */
+    boolean bool = true;
+      // violation below, 'Block comment has incorrect indentation level 6'
+      /*
+       * (block comment should have the same indentation level as line 9)
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example1.java:19]: Please change content of comments to not have violation message. To clear define that is not a hint, it is just example comment.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example2.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// xdoc section -- start
+public class Example2 {
+  public void foo1() {
+    foo2();
+    // it is OK
+  }
+
+  public void foo2() {
+    foo3();
+    //violation below, 'Comment has incorrect indentation level 8, expected is 4'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example2.java:19]: Move this line out of method, you can place it after `}`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example9.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    if (a == 5) {
+      int b;
+      // it is OK
+    } else if (a ==6) {}
+  }
+
+  public void foo43() {
+    try {
+      int a;
+    // violation below, 'Comment has incorrect indentation level 5, expected is 6'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example9.java:23]: Put violation comment above `try`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example8.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    String s2 = "Code complete!";
+    s1.toString().toString().toString();
+    // violation below, 'Comment has incorrect indentation level 14, expected is 4'
+              // It is Not OK1
+    // violation below, 'Comment has incorrect indentation level 9, expected is 4'
+         //It is Not OK2
+    // violation below, 'Comment has incorrect indentation level 7, expected is 4'
+       //It is Not OK3
+    int b = 18;
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example8.java:28]: Please put all violation comments below this line.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example7.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+// xdoc section -- start
+public class Example7 {
+  void testMethod(int a) {
+    switch (a) {
+      case 4:
+        // it is OK
+      case 5:
+      // violation below, 'Comment has incorrect indentation level 5, expected is 6'
+     // It is not OK
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example7.java:18]: Can we merge this example to Example5?

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example6.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.indentation.commentsindentation;
+
+// xdoc section -- start
+public class Example6 {
+  void testMethod() {
+    String breaks = "J"
+        // violation below, 'Comment has incorrect indentation level 4'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example6.java:14]: Move violation comment out of target statement - above `String breaks`.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        int k = 7;
+        break;
+      // it is OK
+      case "3":
+        if (true) {}
+        // violation below, 'Comment has incorrect indentation level 12'
+            //It is not OK
+        break;
+      case "4":
+        break;
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java:28]: Please remove extra cases that don't show anything about behavior of this Check

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        int j = 7;
+        // it is OK
+        break;
+      case "2":
+        int k = 7;
+        break;
+      // it is OK
+      case "3":
+        if (true) {}
+        // violation below, 'Comment has incorrect indentation level 12'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java:24]: Put this comment as training comment for line `case "3":`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java (hunk 2 above)
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java:17]: Please move it above `foo2()` and make it `violation 2 lines above `

**Review by romani — CHANGES_REQUESTED** [review]
Thanks a lot.
This is not easy Check for examples.

Items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java (hunk 2 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example1.java (hunk 3 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example1.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example9.java (hunk 5 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example9.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example8.java (hunk 6 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example8.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example7.java (hunk 7 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example7.java]: Done merged with Example5

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example6.java (hunk 8 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example6.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java (hunk 9 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java (hunk 10 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java:20]: Done###

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java (hunk 2 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example3.java]: Done

**Comment by romani** [comment]
CI failure.

Please practice to run `mvn clean verify ` on local, it helps to catch majority of CI failures before pushing

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java (hunk 10 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java:20]: I meant comment `// violation below, 'Comment`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java (hunk 9 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java]: I still see it

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java]: I still see it

**Comment by romani** [comment]
Please combine example1 and example2 in one example.

Please do following structure of examples:

Default config.
Example1 java.

Changed config to violate only single line comments.
Example1 java (without violations on javadoc like comments)

"Please take a look at the following examples to understand how the check works:"

.... and bellow all examples as they are now.

**Comment by AmitKumarDeoghoria** [comment]
> github

**Comment by AmitKumarDeoghoria** [comment]
> CI failure.
> 
> Please practice to run `mvn clean verify ` on local, it helps to catch majority of CI failures before pushing

Yeah I always run it before pushing any code. But here in this case the errors I am getting was same as violations we are raising. I thought for some reason this indentation check will be ignored in the CI. So what should we do to disable the comment indentation check since we are enabling examples for that only?
Screenshot 2024-11-19 at 11 42 25 AM

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java (hunk 9 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java (hunk 10 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/Example5.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
@AmitKumarDeoghoria , when you have problems to resolve violation , please mention this in comment, I will help.
Without extra comment I treat it as just another missed point and you know how to fix it, just need one more fix.

**Pull request merged**